lib/binaries/standalone_xml.ts

```diff
@@ -0,0 +1,104 @@
+import * as semver from 'semver';
+
+import {Config} from '../config';
+
+import {BinaryUrl} from './binary';
+import {XmlConfigSource} from './config_source';
+
+export class StandaloneXml extends XmlConfigSource {
+  constructor() {
+    super('standalone');
+    this.xmlUrl = Config.cdnUrls()['selenium'];
+  }
+
+  getUrl(version: string): Promise<BinaryUrl> {
+    return this.getXml().then(() => {
+      if (version === 'latest') {
+        return this.getLatestStandaloneVersion();
+      } else {
+        return this.getSpecificStandaloneVersion(version);
+      }
+    });
+  }
+
+  getVersionList(): Promise<string[]> {
+    return this.getXml().then(() => {
+      return this.getStandaloneList();
+    });
+  }
+
+  /**
+   * Get a list of standalone paths available for the configuration OS type and architecture.
+   */
+  private getStandaloneList(): string[] {
+    let versionPaths: string[] = [];
+
+    for (let content of this.xml.ListBucketResult.Contents) {
+      let contentKey: string = content.Key[0];
+
+      // Filter the selenium-server-standalone.
+      if (contentKey.includes('selenium-server-standalone')) {
+        versionPaths.push(contentKey);
+      }
+    }
+    return versionPaths;
+  }
+
+  private getLatestStandaloneVersion(): BinaryUrl {
+    let list = this.getStandaloneList();
+    let standaloneVersion = null;
+    let latest = '';
+    let latestVersion = '';
+    for (let item of list) {
+      // Get a semantic version.
+      let version = item.split('selenium-server-standalone-')[1].replace('.jar', '');
```

Comment:
As with `getLatestChromeDriverVersion` and `getSpecificChromeDriverVersion`, please move parsing/filtering into a helper function